components/brave_rewards/browser/rewards_service_impl_unittest.cc

```diff
@@ -162,4 +162,27 @@ TEST_F(RewardsServiceTest, OnWalletProperties) {
   EXPECT_CALL(*observer(), OnWalletProperties(_, _, _)).Times(0);
 }
 
+TEST_F(RewardsServiceTest, GetTwitchPublisherData) {
+  const std::string& url("https://www.twitch.tv");
+  const std::string& favicon_url("https://www.twitch.tv/favicon.ico");
+  const std::string& publisher_blob(
+    "<figure class=\"tw-avatar tw-avatar--size-36\">"
+    "<div class=\"tw-border-radius-medium tw-overflow-hidden\">"
+    "<img class=\"tw-avatar__img tw-image\" alt=\"");
+  rewards_service()->GetTwitchPublisherData(
+    -1, url, favicon_url, publisher_blob);
+  ASSERT_NO_FATAL_FAILURE(rewards_service()->GetTwitchPublisherData(
+    -1, url, favicon_url, publisher_blob));
+}
+
+TEST_F(RewardsServiceTest, OnTwitchInfo) {
+  const std::string& url("https://www.twitch.tv");
+  const std::string& page_blob(
+    "<figure class=\"tw-avatar tw-avatar--size-36\">"
+    "<div class=\"tw-border-radius-medium tw-overflow-hidden\">"
+    "<img class=\"tw-avatar__img tw-image\" alt=\"");
+  rewards_service()->OnTwitchInfo(page_blob, url);
+  ASSERT_NO_FATAL_FAILURE(rewards_service()->OnTwitchInfo(page_blob, url));
+}
+
```

Comment:
it would be nice for these to test more specific properties of the functionality than just lack of fatal error, but this is a good start! lgtm for now

Reply:
Thanks. Will definitely put better tests on my to-do list.